### PR TITLE
Revert to ReCAPTCHA 2 for Staticman

### LINKED ISSUE #7: BUG: Staticman reCAPTCHA Error
# Bug Report
## Describe the bug

Staticman V2 was not working due to an issue with request limits. Was upgraded to v3 and now it has a reCAPTCHA error even when disabled.

## To Reproduce
1. Submit a comment via Staticman

## Expected behavior

Comment should successfully post
##Screenshots

## Environment
 - **Hugo Version:** 0.50
 - **Browser and Version:** 
 - **Platform:** 
 - **Link to Project: **

## Additional context

See eduardoboucas/staticman#243

### PR DESCRIPTION
**Pull request by pacollins**
## Description

ReCAPTCHA reverted back to v2 Checkbox for Staticman

## Motivation and Context

Closes #7 

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] I have read the [Contributing Document](https://github.com/pacollins/hugo-future-imperfect-slim/blob/master/.github/CONTRIBUTING.md).
- [X] My code follows the [code style](https://github.com/pacollins/hugo-future-imperfect-slim/blob/master/.github/CONTRIBUTING.md#Style-Guide) of this project.
- [ ] My change requires a change to the [documentation](https://github.com/pacollins/hugo-future-imperfect-slim/wiki).
- [ ] I have updated the documentation, including `theme.toml`, accordingly.

### CODE REVIEW

File: layouts/partials/head.html [hunk 1]
```diff
@@ -22,7 +22,7 @@
       <link rel="stylesheet" href="{{ . | relURL }}">
     {{ end }}
   {{ end }}
-  {{ if .Site.Params.staticman.recaptcha.recaptcha }}<script src="https://www.google.com/recaptcha/api.js?render={{ .Site.Params.staticman.recaptcha.sitekey }}"></script>{{ end }}
+  {{ if .Site.Params.staticman.recaptcha.recaptcha }}<script src='https://www.google.com/recaptcha/api.js'></script>{{ end }}
```
Comment (VincentTam) [inline layouts/partials/head.html]: You can drop the `recaptcha` field under `[params.staticman.recaptcha]` in the site config file to keep things slim.

File: layouts/post/staticman.html [hunk 2]
```diff
@@ -9,23 +9,12 @@ <h2>Say something</h2>
     <input required name="fields[email]" type="email" placeholder="Your email address (Required for Gravatar)">
     <textarea required name="fields[body]" placeholder="Your message. Feel free to use Markdown (Google 'Markdown Cheat Sheet')." rows="10"></textarea>
     {{ if .Site.Params.staticman.recaptcha.recaptcha }}
```
Comment (VincentTam) [inline layouts/post/staticman.html]: You may consider using `{{ with .Site.Params.staticman.recaptcha }}` to avoid repetitions.

File: static/css/main.min.css [hunk 3]
```diff
@@ -1 +1,7 @@
...
...are a{margin-bottom:.5em;padding:.5em;width:4em}@media (min-width: 768px){#social-share a{margin-right:.25em}}#social-share a i{font-size:1em}#social-share p{display:none}.hidden{display:none !important}.visuallyhidden{border:0;clip:rect(0 0 0 0);height:1px;margin:-1px;overflow:hidden;padding:0;position:absolute;width:1px;white-space:nowrap}.visuallyhidden.focusable:active,.visuallyhidden.focusable:focus{clip:auto;height:auto;margin:0;overflow:visible;position:static;width:auto;white-space:inherit}.invisible{visibility:hidden}.clearfix:before,.clearfix:after{content:" ";display:table}.clearfix:after{clear:both}@media print{*,*:before,*:after{background:transparent !important;color:#000 !important;-webkit-box-shadow:none !important;box-shadow:none !important;text-shadow:none !important}a,a:visited{text-decoration:underline}a[href]:after{content:" (" attr(href) ")"}abbr[title]:after{content:" (" attr(title) ")"}a[href^="#"]:after,a[href^="javascript:"]:after{content:""}pre{white-space:pre-wrap !important}pre,blockquote{border:1px solid #999;page-break-inside:avoid}thead{display:table-header-group}tr,img{page-break-inside:avoid}p,h2,h3{orphans:3;widows:3}h2,h3{page-break-after:avoid}}
```
Comment (VincentTam) [inline static/css/main.min.css]: `socnet` has been removed.  This leads to CSS errors.

Fish demo CSS error

**Comment by VincentTam** [comment]
Staticman with reCAPTCHA worked on my demo site: https://framagit.org/vincenttam/fish-demo.  \@staticmanlab1 managed to push again the `master` branch of the corresponding repo.

File: layouts/partials/head.html (hunk 1 above)
Comment (pacollins) [reply layouts/partials/head.html]: Replaced with `{{ if and .Site.Params.staticman.recaptcha.siteKey .Site.Params.staticman.recaptcha.encryptedKey }}`.

Comment (pacollins) [reply layouts/partials/head.html]: Correction, using `{{ with .Site.Params.staticman.recaptcha }}` which only returns `true` if both fields are present.

File: layouts/post/staticman.html (hunk 2 above)
Comment (pacollins) [reply layouts/post/staticman.html]: Replaced with `{{ with .Site.Params.staticman.recaptcha }}`.

File: static/css/main.min.css (hunk 3 above)
Comment (pacollins) [reply static/css/main.min.css]: Accidental merge error, reinstating.

File: layouts/post/staticman.html (hunk 2 above)
Comment (pacollins) [reply layouts/post/staticman.html]: Replaced with `{{ if and .Site.Params.staticman.recaptcha.siteKey .Site.Params.staticman.recaptcha.encryptedKey }}`.

**Comment by pacollins** [comment]
Now that I have made the change and pushed it. I am torn.

`...recaptcha.recaptcha` served as a way to simply turn off the feature without deleting the keys. Theoretically, you could just comment out the keys now I suppose.

If we are doing this, should we change [`...staticman.staticman`](https://github.com/pacollins/hugo-future-imperfect-slim/blob/f0ec5ac57e6765ef2cf1cb4a2f585bfc1c1c5c6b/layouts/post/comments.html#L5) to a similar system of `{{ if and ...}}`?

**Comment by VincentTam** [comment]
> Now that I have made the change and pushed it. I am torn.

Thanks for editing.  It's Sunday, so you may take a rest and lemme do the testing and reviewing work for you.

> ...recaptcha.recaptcha served as a way to simply turn off the feature without deleting the keys. Theoretically, you could just comment out the keys now I suppose.

Even though this theme is called "Future Imperfect", when we look into the past, that's really a breaking change.  But what's broken?  Here's the only case.

```
    [params.staticman.recaptcha]
      recaptcha = false
      siteKey           = "<nonempty site key>" # Site Key
      encryptedKey      = "<nonempty encrypted key>"
```

Here're the repos using this theme with `recaptcha = false`: https://github.com/search?o=asc&q=theme+%3D+%22hugo-future-imperfect-slim%22+AND+%22recaptcha+%3D+false%22+filename%3Aconfig.toml+fork%3Atrue&s=indexed&type=Code.  I haven't (and perhaps won't) check all of them, but from the samples that I've viewed, *all* of them has simply left this section untouched—perhaps due to error #7.

To sum up, we're breaking a broken stuff (so as to reassemble and fix it), so we don't need to worry about this.



> If we are doing this, should we change `...staticman.staticman` to a similar system of `{{ if and ...}}`? The new method would be backwards compatible in theory (assuming people only fill in the information if they want to use the feature).
> 
> https://github.com/pacollins/hugo-future-imperfect-slim/blob/f0ec5ac57e6765ef2cf1cb4a2f585bfc1c1c5c6b/layouts/post/comments.html#L5-L9

Lemme do that for you.  From the last part, we can forget about the backward compatibility.

Upcoming: some Staticman changes out of the scope of this PR.

**Comment by VincentTam** [comment]
:pray: Thanks for your great work this w.e.  :microscope: I've just tested it again on [my test site on Framagit :globe_with_meridians:](https://vincenttam.frama.io/fish-demo) served by the Framagit instance [@staticmanlab1](https://framagit.org/staticmanlab1).  It worked !!!

# :memo: Test report [TL;DR]

1. Squash merged recent commits into [my `dev` branch](https://github.com/VincentTam/hugo-future-imperfect-slim/tree/dev).

    Compare my `dev` branch with your `Revert-staticman-recaptcha` branch: https://github.com/pacollins/hugo-future-imperfect-slim/compare/Revert-staticman-recaptcha..VincentTam:dev?expand=1

    SHA-1 hash of my `dev` branch: 5a3e7561

2. Updated the submodule of my test site.

    SHA-1 hash of the `master` branch: [`18df4ac`](https://framagit.org/VincentTam/fish-demo/commit/18df4acb2ce7b6eaad48ae04797192b3feb6968a)

3. Submitted a comment at [a sample post](https://vincenttam.frama.io/fish-demo/blog/migrate-from-jekyll/).
4. Redirected to the post.  \@staticmanlab1 commited directly into the `master` branch.

    SHA-1 hash of the `master` branch: [`19fb99d`](https://framagit.org/VincentTam/fish-demo/commit/19fb99d4c801137b49a6823ef3aec9e3898b9622).  Its parent is the commit mentioned in step 2.  Note that the `config.toml` has not yet been changed.  This shows the backward compatibility of this PR at ff52ad0.

5. Git pulled the Staticman comment in step 3 into my local repo.
6. Removed the lines `recaptcha = true` and `staticman = true` in `config.toml`.

    ```diff
    diff --git a/config.toml b/config.toml
    index 51fc23e..e3a379e 100644
    --- a/config.toml
    +++ b/config.toml
    @@ -92,7 +92,6 @@ disableLanguages        = [""]
         # Sets Statiman to be active
         # If using GitHub, go to https://github.com/apps/staticman-net
         # If using GitLab, just add the GitLab bot, NO need to hit `/connect/v3/...`
    -    staticman           = true
         # Sets the location for Staticman to connect to
         api                 = "staticman-frama.herokuapp.com"  # without trailing slash
         gitProvider         = "gitlab"  # either "github" or "gitlab"
    @@ -101,7 +100,6 @@ disableLanguages        = [""]
         branch              = "master"
     
         [params.staticman.recaptcha]
    -      recaptcha         = true
           siteKey           = "6Lcv8G8UAAAAAEqV1Y-XEPum00C_DxhD6O--qkFo" # Site Key
           encryptedKey      = "GLGBV5a+DHijkgBrPXRVvlK2CUrgyo0vOgpnVFTX76VUsTtrxqK8X5mJ9IjXiZbSmEwpflr6il1aLPueoV38nv9ZBIbPNUzj1xmt1CQ9HUOQFpB+n22HlwKJPqj29X8xkgCKnR190WK/KpPau6rF/PYzQn/eI+6hEEcCfOWUm9iQqUSk9vpCTtFD7LnFtxd+TjCxfY8R0uFpvs141aOAv/8wcR3ZzOjoWx3adAQkAgu6QLC1SVBQVnqDKiBrw4Tys/P8LhJE2Ul28tItI7XH5ZUpt8AV/38t5v+6cMEAk3sUr1VqnrPfjnQrLysXQlyeHMqqV48IZLLBWuMb0OqGog=="
     ```

    SHA-1 hash of this commit: [`62c5562`](https://framagit.org/VincentTam/fish-demo/commit/62c5562beee493f0173e3a8a2d587dd3e36886f9)

7. Submitted a comment again at the post mentioned in step 3.
8. \@staticmanlab1 received the comment, and regenerated the site at commit [`3c1d2d9`](https://framagit.org/VincentTam/fish-demo/commit/3c1d2d97d0bd839598c22a930dafc947d7037832), whose parent is the commit mentioned in step 6.
